src/nanotron/models/llama.py

```diff
@@ -210,6 +213,8 @@ def forward(
         # TODO(kunhao): flash attn's causal means that the query can only attend to the keys before it. This is not
         # what we want if we are using kv cache. This is a hack as we always have q_length == 1 when using kv cache.
         causal = False if q_sequence_mask.shape[1] == 1 else True
+        # NOTE: this scale is for µTransfer
+        softmax_scale = 1 / query_states.shape[-1]
```

Comment:
it's alrdy the case

Reply:
no need to specify it

Reply:
nah. by default, we do sqrt(1/d_head), but µTransfer do 1/d_head (no sqrt)